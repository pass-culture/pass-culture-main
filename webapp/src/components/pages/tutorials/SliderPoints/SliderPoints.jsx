import React from 'react'
import PropTypes from 'prop-types'

const SliderPoints = ({ currentStep, maxStep }) => {
  const points = () => {
    const pointsToDisplay = []

    for (let step = 0; step < maxStep; step++) {
      const isFilled = step === currentStep

      pointsToDisplay.push(
        <div
          className={`slider-point ${isFilled ? 'filled' : ''}`}
          key={step}
        />
      )
    }
    return pointsToDisplay
  }

  return points()
}

SliderPoints.propTypes = {
  currentStep: PropTypes.number.isRequired,
  maxStep: PropTypes.number.isRequired,
}

export default SliderPoints
