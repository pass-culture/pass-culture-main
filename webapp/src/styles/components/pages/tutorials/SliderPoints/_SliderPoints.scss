.slider-point {
  background-color: $white;
  border-radius: rem(8px);
  height: rem(8px);
  margin: 0 rem(5px);
  opacity: 1;
  width: rem(8px);

  &.filled {
    height: rem(14px);
    width: rem(14px);
  }
}
