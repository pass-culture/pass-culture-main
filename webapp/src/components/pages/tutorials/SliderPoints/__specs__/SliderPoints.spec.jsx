import { shallow } from 'enzyme'
import React from 'react'

import SliderPoints from '../SliderPoints'

describe('components | SliderPoints', () => {
  it('should have as much bullet points as there are elements and current step point should be filled', () => {
    // given
    const maxStep = 4
    const currentStep = 2

    const props = {
      maxStep,
      currentStep,
    }

    // when
    const wrapper = shallow(<SliderPoints {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })
})
