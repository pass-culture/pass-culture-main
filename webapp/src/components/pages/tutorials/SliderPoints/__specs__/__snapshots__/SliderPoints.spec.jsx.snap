// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | SliderPoints should have as much bullet points as there are elements and current step point should be filled 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderPoints
    currentStep={2}
    maxStep={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "0",
    "nodeType": "host",
    "props": Object {
      "className": "slider-point ",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "0",
      "nodeType": "host",
      "props": Object {
        "className": "slider-point ",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "host",
      "props": Object {
        "className": "slider-point ",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    Object {
      "instance": null,
      "key": "2",
      "nodeType": "host",
      "props": Object {
        "className": "slider-point filled",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    Object {
      "instance": null,
      "key": "3",
      "nodeType": "host",
      "props": Object {
        "className": "slider-point ",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
